utils/sambanova_endpoint.py

```diff
@@ -1070,24 +1076,26 @@ def _handle_nlp_predict_stream(
             }
 
             if chunk['status_code'] == 200 and chunk.get('error'):
-                chunk['result'] = {'responses': [{'stream_token': ''}]}
+                raise RuntimeError(
+                    f"Sambanova /complete call failed with status code " f"{chunk['status_code']}." f"{chunk}."
+                )
+                
             if chunk['status_code'] != 200:
-                error = chunk.get('is_error')
-                optional_details = chunk.get('completion')
-                if error:
-                    optional_details = error.get('details')
-                    raise ValueError(
-                        f"Sambanova /complete call failed with status code "
-                        f"{chunk['status_code']}.\n"
-                        f"Details: {optional_details}\n"
-                    )
-                else:
-                    raise RuntimeError(
-                        f"Sambanova /complete call failed with status code " f"{chunk['status_code']}." f"{chunk}."
-                    )
-            text = json.loads(chunk['data'])['stream_token']
-            generated_chunk = GenerationChunk(text=text)
-            yield generated_chunk
+                raise RuntimeError(
+                    f"Sambanova /complete call failed with status code " f"{chunk['status_code']}." f"{chunk}."
+                )
+                
+            # check if the response is a final '[DONE]' response
+            if chunk['data']!="[DONE]" :
```

Comment:
Could you explain why this condition is `!= DONE`. It sounds like when it is not finished but in the downstream condition you're yielding the generated text. Is it because You haven't reached final streaming chunk?

Reply:
this is checking is not a final event in the ss-events stream, when this is the case there is not any data sent only the string ["DONE"] in event data,
more explanatory comment added